feugy/tabulous
# feat: invites friends in games (and only friends)

### :book: What's in there?

Young players' safety is a top-priority concern. Now it is only allowed to invite friends in games, which gives more fine-grained control on who can offer you new games, and who you'll be playing with.
Therefore:
1. the game page top-left menu does not have an "invite" option to open the invite dialogue
1. no more invite dialogue when creating a new lobby, 
1. no more invite button in the current lobby card
1. when being in a game or lobby, the friends side panel has a player/attendee list on top of the friends list

Are included here:

- feat(server): supports inviting multiple guests, requires them to be friends
- feat(web): invites guests from friend list, removes `InvitePlayerDialogue`
- feat(web): adds button to request friendship with a player of the current game that is not part of the current friend list
- chore(web): uses Playwright idomatic locators
- chore(web): updates vitest

### :test_tube: How to test?

1. Open a new game or lobby
   > the "friends" tab appears on the right-hand aside panel
3. Click on the friends tab icon to expand it
   > your friend list appears
4. Click on one (or more) of your friends
   > they are selected, and a new "invite to game/lobby" button appears at the list's bottom
5. Click on the invite button
   > the player's tab appears with its content

[Capture vidéo du 18-02-2023 18:23:07.webm](https://user-images.githubusercontent.com/186268/219879881-a8dbcf2a-ad53-43a8-ba2a-fcfa6905b49a.webm)
